app/models/affair.rb

```diff
@@ -154,6 +154,10 @@ def affairable_types
       @affairable_types ||= %w[SaleGap PurchaseGap Sale Purchase Payslip IncomingPayment OutgoingPayment Regularization DebtTransfer].freeze
     end
 
+    def affairable_types_with_third
```

Comment:
Why not make this a bit more dynamic:
```ruby
@affairable_types.map(&:constantize).select { |model| model.respond_to? :third_id }
```

Reply:
I even wonder if we do need the method at all since it's only used once. Maybe a variable instead? What do you think?